### ISSUE TITLE
Section disappears from view after a few blocks are destroyed

### ISSUE THREAD

**Issue by teoscura** [ferrumc-rs/ferrumc#148]
### Describe the Bug

As you interact with the world, punching any block will lead to the disappearance of the section those blocks belonged to after a few seconds.

### To Reproduce

Load any world. punch a few blocks, wait a few seconds.

### Expected Behaviour

It is the expected behaviour, as the server isn't validating any of the block changes happening on the client, the client therefore panics and removes the whole section.

### Screenshots / Videos

![image](https://github.com/user-attachments/assets/fe6ac323-ba1f-44b4-80a2-39776e5add06)

### FerrumC Version

v0.1.0

### Rust Version

1.76

### Server Logs

``cargo run
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.23s
     Running `target/debug/ferrumc`
ChunkReceiver: CB50A085A553777C
StreamWriter: FD92C34816924937
2024-12-30T21:10:05.839302Z  INFO ferrumc: Starting server...
2024-12-30T21:10:05.840238Z DEBUG ferrumc_net::server: Trying to bind to 0.0.0.0:25565
2024-12-30T21:10:05.840627Z  INFO ferrumc_world: Using gzip compression algorithm
2024-12-30T21:10:05.842025Z  INFO sys{name=Chunk Fetcher}: ferrumc::systems::chunk_fetcher: Chunk fetcher system started
2024-12-30T21:10:05.842025Z  INFO sys{name=tcp}: ferrumc::systems::tcp_listener_system: Server is listening on [0.0.0.0:25565]
2024-12-30T21:10:05.842105Z DEBUG sys{name=tcp}: ferrumc::systems::tcp_listener_system: Accepting connection
2024-12-30T21:10:05.842093Z  INFO sys{name=Chunk Sender}: ferrumc::systems::chunk_sender: Chunk sender system started
2024-12-30T21:10:05.842115Z  INFO sys{name=keep_alive}: ferrumc::systems::keep_alive_system: Started keep_alive
2024-12-30T21:10:05.842218Z  INFO sys{name=keep_alive}: ferrumc::systems::keep_alive_system: Online players: 0
2024-12-30T21:10:20.842655Z  INFO sys{name=keep_alive}: ferrumc::systems::keep_alive_system: Online players: 0
2024-12-30T21:10:21.214355Z DEBUG sys{name=tcp}: ferrumc::systems::tcp_listener_system: Accepting connection
2024-12-30T21:10:21.289042Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc::packet_handlers::login_process: Handling login start event for user: Scurax, uuid: 169589585263843882894824967731405991537
2024-12-30T21:10:21.319298Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net::packets::incoming::server_bound_plugin_message: Received plugin message: ServerBoundPluginMessage { channel: "minecraft:brand", data: [7, 118, 97, 110, 105, 108, 108, 97] }
2024-12-30T21:10:21.319366Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net::packets::incoming::server_bound_plugin_message: Client brand: vanilla
2024-12-30T21:10:21.319866Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net::packets::incoming::client_information: Received client information: ClientInformation {
    locale: "en_us",
    view_distance: 12,
    chat_mode: Enabled,
    chat_colors: true,
    displayed_skin_parts: 127,
    main_hand: Right,
    enable_text_filtering: true,
    allow_server_listings: true,
}
2024-12-30T21:10:21.342259Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net::packets::incoming::server_bound_known_packs: Received known packs: ServerBoundKnownPacks {
    packs: LengthPrefixedVec {
        length: VarInt {
            val: 0,
            len: 1,
        },
        data: [],
    },
}
2024-12-30T21:10:21.766708Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x00 in state: play
wa2024-12-30T21:10:23.761020Z  INFO sys{name=ticking}: ferrumc::systems::ticking_system: Running behind by 1.686045441s
2024-12-30T21:10:23.783442Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x08 in state: play
2024-12-30T21:10:33.063969Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x0A in state: play
2024-12-30T21:10:33.759454Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x0A in state: play
2024-12-30T21:10:35.843937Z  INFO sys{name=keep_alive}: ferrumc::systems::keep_alive_system: Online players: 1
2024-12-30T21:10:36.488544Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x24 in state: play
2024-12-30T21:10:36.635398Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x24 in state: play
2024-12-30T21:10:36.983188Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x24 in state: play
2024-12-30T21:10:46.538238Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net: No packet found for ID: 0x24 in state: play
2024-12-30T21:10:50.844708Z  INFO sys{name=keep_alive}: ferrumc::systems::keep_alive_system: Online players: 1
2024-12-30T21:11:05.846469Z  INFO sys{name=keep_alive}: ferrumc::systems::keep_alive_system: Online players: 1
2024-12-30T21:11:17.640455Z DEBUG sys{name=tcp}:conn{addy=127.0.0.1:35778}: ferrumc_net::connection: Connection closed for entity: 0 
``
### Additional Context

None applicable.

### Is there an existing issue for this?

- [X] I have searched current existing issues before creating this.
- [ ] I checked all the boxes without reading them
- [X] I have provided all necessary and relevant details, including logs, reproduction steps, etc.

**Comment by GStudiosX2** [comment]
I talked about this on discord a few days back I forgot to make a issue so this is technically known I believe that it's because we don't properly set the block count in the chunk sections at the moment

**Comment by ReCore-sys** [comment]
Yeah I am aware of this, it's due to very incorrectly setting the amount of non-air blocks in a chunk. Should get fixed when I finish the block editing.
